_posts/2016-12-02-one-year-on-gcp.md

```diff
@@ -0,0 +1,62 @@
+
+#### Container Engine
+
+Container Engine is GCPs answer for hosting linux containers. It’s powered by Kubernetes which is, as of writing, the de facto standard for scheduling and running linux containers in production. On GCP I view Container Engine as the middle ground between Compute and App Engine. Where I believe you get the best tradeoff between operational overhead and flexibility. With Kubernetes you can do interesting things as [bundle databases or other services together to increase performance](http://labs.unacast.com/2016/11/22/high-performance-read-api/) which is impossible in App Engine. However, you have to worry about updating your Kubernetes cluster and keeping the nodes healthy and happy. Adding some operational complexity, work and time spent on not adding features.
+
+#### Dataflow
+
+Dataflow all the things! Dataflow is GCPs next generation MapReduce. It has streaming and batch capabilities. Dataflow is so good that we try to use it every time we need to process a non-trivial amount of data or we just need to run continiously running workers. As of writing Dataflow only has a SDK for Java. And Java isn't necessarily the natural language for defining and working with data pipelines. Needless to say we started to look for non-official SDKs that could suit our needs. We found [Datasplash](https://github.com/ngrunwald/datasplash), a Dataflow wrapper written in Clojure, and are quite satisfied with it. Clojure  syntax and functional approach works very well when defining data processing pipelines. We're currently pretty happy with Datasplash/Clojure, at the time of writing we're running Dataflow pipelines written in Java and Clojure. Time will show if this is the right tool. A caveat with Dataflow on GCP is that it uses Google Compute Engines under the hood. And that means the quota limits for virtual machines can be a show stopper. Make sure to always have large enough limits while you're evolving your platform.
+
+## The not so good
+
+#### Stackdriver
+Stackdrivers monitors sucks. At least from my experience of it. Monitoring is hard from the get go. Because its hard to know what to monitor and setting up real good monitors. If the monitor is too verbose and sensitive nobody cares when an alert is triggered, and if it’s too sensitive errors in production will go unnoticed. In my opinion [setting up custom metrics](https://cloud.google.com/monitoring/custom-metrics/creating-metrics) in Stackdriver is a horrible experience. And that is why we use [Datadog](https://datadoghq.com) for monitoring services and setting up dashboards. To be fair, Stackdriver has some good components too, especially if you’re using AppEngine. Stackdrivers Trace functionality is awesome for tracking what is slow in your application. And the logs module are easy to use and query for interesting info. My experience is that these two modules works really great out of the box.
```

Comment:
The five first sentences is a bit stumbling. Maybe remove "at least form my experience", since this is an experience report. We're allowed to have opinions, and we don't need to sugar coat anything.

Also we're saying that the monitor is 'too verbose and sensitive' or 'too sensitive'.